tests/unit/io/WKTReaderTest.cpp

```diff
@@ -473,4 +473,86 @@ void object::test<24>
     ensure_equals(geom->getNumGeometries(), 3u);
 }
 
+// Read a CircularString
+template<>
+template<>
+void object::test<25>
+()
+{
+    auto geom = wktreader.read("CIRCULARSTRING (0 0, 1 1, 2 0)");
+
+    ensure_equals(geom->getGeometryTypeId(), geos::geom::GEOS_CIRCULARSTRING);
+    ensure_equals(geom->getNumPoints(), 3u);
+}
+
+// Read a CompoundCurve
+template<>
+template<>
+void object::test<26>
+()
+{
+    auto geom = wktreader.read("COMPOUNDCURVE (CIRCULARSTRING (0 0, 1 1, 1 0), (1 0, 0 1))");
```

Comment:
I can't remember if it's a PostGIS Extension or OGC one, but, you can write 'LINESTRING' inside a compoundcurve:

At least, PostGIS accepts:
'COMPOUNDCURVE (CIRCULARSTRING (0 0, 1 1, 1 0), LINESTRING (1 0, 0 1))' as
'COMPOUNDCURVE (CIRCULARSTRING (0 0, 1 1, 1 0), (1 0, 0 1))'

Reply:
Thanks for pointing this out. I added tests to verify this variant is handled.